tests/shared/src/test/scala/cats/tests/EitherTSuite.scala

```diff
@@ -655,6 +655,35 @@ class EitherTSuite extends CatsSuite {
     }
   }
 
+  test("biSemiflatTap does not change the return value") {
+    type TestEffect[A] = State[List[Int], A]
+    forAll {
+      (eithert: EitherT[TestEffect, String, Int],
+       fa: String => TestEffect[Int],
+       fb: Int => TestEffect[Int],
+       initial: List[Int]
+      ) =>
+        assert(eithert.biSemiflatTap(v => fa(v), v => fb(v)).value.runA(initial) === (eithert.value.runA(initial)))
+    }
+  }
+
+  test("biSemiflatTap  consistent with leftSemiflatTap and semiFlatTap") {
+    type TestEffect[A] = State[List[Int], A]
+    forAll {
+      (eithert: EitherT[TestEffect, String, Int],
+       fa: String => TestEffect[Int],
+       fb: Int => TestEffect[Int],
+       initial: List[Int]
+      ) =>
+        assert(
+          eithert
+            .biSemiflatTap(fa, fb)
+            .value
+            .runS(initial) === (eithert.leftSemiflatTap(fa).semiflatTap(fb).value.runS(initial))
+        )
+    }
+  }
```

Comment:
Looks good to me, but I would appreciate if you could clean this block up for better readability:
- Remove extra space between `biSemiflatTap` and `consistent`:
  ```
  test("biSemiflatTap  consistent with leftSemiflatTap and semiFlatTap")
  ```
- Either reformat for better clarity:
  ```
  assert(
    eithert.biSemiflatTap(fa, fb).value.runS(initial) ===
      eithert.leftSemiflatTap(fa).semiflatTap(fb).value.runS(initial))
  ```
  **OR** consider replacing `assert` with `assertEquals` which would look even better and provide better output if a case of assertion error.

I'm going to approve anyway though.

Reply:
9d2fa4d24974487b6d208f828ef91e199df8b445